src/Compilers/CSharp/Portable/BoundTree/NullabilityRewriter.cs

```diff
@@ -8,5 +14,58 @@ protected override BoundExpression VisitExpressionWithoutStackGuard(BoundExpress
         {
             return (BoundExpression)Visit(node);
         }
+
+        public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
+        {
+            return VisitBinaryOperatorBase(node);
+        }
+
+        public override BoundNode VisitUserDefinedConditionalLogicalOperator(BoundUserDefinedConditionalLogicalOperator node)
+        {
+            return VisitBinaryOperatorBase(node);
+        }
+
+        private BoundNode VisitBinaryOperatorBase(BoundBinaryOperatorBase binaryOperator)
+        {
+            // Use an explicit stack to avoid blowing the managed stack when visiting deeply-recursive
+            // binary nodes
+            var stack = ArrayBuilder<BoundBinaryOperatorBase>.GetInstance();
+            BoundBinaryOperatorBase currentBinary = binaryOperator;
+
+            do
+            {
+                stack.Push(currentBinary);
+                currentBinary = currentBinary.Left as BoundBinaryOperatorBase;
+            } while (currentBinary != null);
```

Comment:
>} while (currentBinary != null); [](start = 12, length = 32)

Is placing ```while``` on the same line as the closing brace (```}```) follows our coding style?

Reply:
Well, the formatting analyzer doesn't complain about it, but by allman bracing styles it probably should.

---
In reply to: [271539092](https://github.com/dotnet/roslyn/pull/34429#discussion_r271539092) [](ancestors = 271539092)